File: workspace/python/playground/playground-fastapi/quick-startup/utils/http/response.py
# coding: UTF-8
"""
@software: PyCharm
@author: Lionel Johnson
@contact: https://fairy.host
@organization: https://github.com/FairylandFuture
@datetime: 2025-09-11 21:29:16 UTC+08:00
"""

import typing as t
from dataclasses import dataclass


@dataclass()
class Response:
    code: t.Optional[int] = None
    message: t.Optional[str] = None
    data: t.Optional[t.Any] = None

    def __post_init__(self):
        if not self.code:
            self.code = 200
        if not self.message:
            self.message = "success"
        if not self.data:
            self.data = {}

    def to_dict(self) -> dict:
        return {
            "code": self.code,
            "message": self.message,
            "data": self.data,
        }

File: workspace/java/web/quickstart/src/main/java/host/fairy/model/http/Response.java
/*****************************************************
 * @software: IntelliJ IDEA
 * @author: Lionel Johnson
 * @contact: https://fairy.host
 * @organization: https://github.com/FairylandFuture
 * @datetime: 2026-03-01 22:06:02 UTC+08:00
 ****************************************************/
package host.fairy.model.http;

import lombok.Builder;
import lombok.Data;

import java.io.Serializable;

/**
 * @author Lionel Johnson
 * @version 1.0
 */
@Data
@Builder
public class Response<T> implements Serializable {
    
    private int code;
    private String message;
    private T data;
    
    public static <T> Response<T> success(T data) {
        return Response.<T>builder()
                .code(200)
                .message("OK")
                .data(data)
                .build();
    }
}
